src/types/OpenAPI3.ts

```diff
@@ -9,13 +9,27 @@ export interface OpenAPI3Schemas {
 }
 
 export interface OpenAPI3Paths {
-  [path: string]: {
-    [method: string]: OpenAPI3Operation | Parameter[];
-  };
+  [path: string]: Partial<OpenAPI3PathItem>;
 }
 
+export type OpenAPI3PathItem = {
+  $ref: string;
+  summary: string;
+  description: string;
+  get: OpenAPI3Operation;
+  put: OpenAPI3Operation;
+  post: OpenAPI3Operation;
+  delete: OpenAPI3Operation;
+  options: OpenAPI3Operation;
+  head: OpenAPI3Operation;
+  patch: OpenAPI3Operation;
+  trace: OpenAPI3Operation;
+  parameters: Parameter[];
+};
```

Comment:
nice. I wonder why you didn't made all the keys optional and instead used the `Partial<>` method above? Is there any benefit in that approach? Because I don't see the `OpenAPI3PathItem` type by itself to ever be valid.

Reply:
This is a good point and I don't think I have a good enough reason to push back on it. I will update the PR with this change.

Reply:
This is updated with the [last push here](https://github.com/drwpow/openapi-typescript/pull/366/files/9b31327b90c39d899b788a974e0f7f80d6f045fa..efed10f2a66e1cf1385de147ed01b784e8995b72)

Reply:
cheers 👍🏼

Reply:
This change looks good to me! In general it’s probably best to err on the side of making too many properties optional. In my experience even if the official OpenAPI spec marks something as required, almost all generators/parsers will still work, so this library should too (I think of it as a loose spec).

Unrelated additional context: we only have partial types in this repo, and only the ones this library knows about, for 2 reasons:

1. I’ve found some projects with OpenAPI TypeScript types but the ones I looked at are incomplete or inaccurate in some way (if some exist that work well I’d be open to it)
2. In some instances we may want to deviate from the spec, such as making something optional even if it’s not, so we can allow more generation.

Sorry; a very long way to say “LGTM” 😄